src/opnsense/www/js/opnsense.js

```diff
@@ -94,7 +94,12 @@ function getFormData(parent) {
                     }
                 } else {
                     // regular input type
-                    node[keypart] = sourceNode.val();
+                    if (sourceNode.hasClass("json-data")) {
+                        // deserialize the field content - used for JS maintained fields
+                        node[keypart] = JSON.parse(sourceNode.val());
```

Comment:
better to use a data tag for this, should avoid json serialisation:

`node[keypart] = sourceNode.data('json');`

(We're doing something similar for another project)